core/cmd/remote_client.go

```diff
@@ -66,8 +66,29 @@ func (cli *Client) CreateServiceAgreement(c *clipkg.Context) error {
 }
 
 // CreateExternalInitiator adds an external initiator
+//
+// TODO: Investigate reducing the number of return statements
+// TODO: Increase code coverage
 func (cli *Client) CreateExternalInitiator(c *clipkg.Context) error {
-	resp, err := cli.HTTP.Post("/v2/external_initiators", nil)
+	if c.NArg() != 2 {
+		return cli.errorOut(errors.New("create expects two arguments: a name and a url"))
+	}
+
+	var request models.ExternalInitiatorRequest
+	request.Name = c.Args().Get(0)
+	if url, err := models.NewWebURL(c.Args().Get(1)); err == nil {
```

Comment:
You can just use `url.ParseRequestURI`, `models.URL` is just a special type used for JSON.